### PR TITLE
Remove waitbars

### PR DESCRIPTION
**Pull request by lemmerelassal**
*(Note: You may replace [ ] with [X] to check the box)*

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [contributing](https://github.com/opencobra/cobratoolbox/blob/documentation/.github/CONTRIBUTING.md) document
- [X] Followed the [code styleguide](https://github.com/opencobra/cobratoolbox/blob/documentation/.github/CONTRIBUTING.md#code-styleguide)
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/opencobra/cobratoolbox/pulls) for the same update/change
- [X] Written platform-independent code
- [X] Used `pwd` to get the current directory
- [X] Used `filesep` for paths (e.g., `['myPath' filesep 'myFile.m']`)
- [X] Made sure that computational results are reproducible

#### Previous output

[*Include here information of the current version of The COBRA Toolbox*]

#### Output of PR-merged version

[*Include here information of modified version of The COBRA Toolbox after your proposed PR is accepted*]

###  Documentation updates

[*Include here information on how the documentation has been updated*]

**I hereby confirm that I have:**

- [X] Documented your code based on the [Documentation styleguide](#documentation-and-comments-styleguide).
- [X] Documented the `Input` and `Output` arguments
- [X] Followed the guidelines to automatically generate the documentation

###  Tests tailored to test the PR

[*Include here information on how new tests and/or how existing tests have been adapted*]

**I hereby confirm that I have:**

- [X] Written a sensible test according to the [Test styleguide](#test-styleguide)
- [X] Tested your PR locally prior to submission
- [X] Checked that all tests pass
- [X] Tested the code on Linux

### CODE REVIEW

File: src/analyzeRxns.m [hunk 1]
```diff
@@ -18,9 +18,11 @@
 
 %find all product producers
 makesProd = [];
-h = waitbar(0,['finding ',product,' producing strains']);
+%h = waitbar(0,['finding ',product,' producing strains']);
```
Comment (syarra) [inline src/analyzeRxns.m]: Remove the commented line of code. Not needed!

File: src/analyzeRxns.m [hunk 2]
```diff
@@ -18,9 +18,11 @@
 
 %find all product producers
 makesProd = [];
-h = waitbar(0,['finding ',product,' producing strains']);
+%h = waitbar(0,['finding ',product,' producing strains']);
+h = showprogress(0,['finding ',product,' producing strains']);
 for i = 1:length(listProducts)
-    waitbar(i/length(listProducts),h);
+    %waitbar(i/length(listProducts),h);
```
Comment (syarra) [inline src/analyzeRxns.m]: same here!

File: src/analyzeRxns.m [hunk 3]
```diff
@@ -18,9 +18,11 @@
 
 %find all product producers
 makesProd = [];
-h = waitbar(0,['finding ',product,' producing strains']);
+%h = waitbar(0,['finding ',product,' producing strains']);
+h = showprogress(0,['finding ',product,' producing strains']);
```
Comment (syarra) [inline src/analyzeRxns.m]: Please follow the coding style guide: `h = showprogress(0, ['finding ', product, ' producing strains']);`

File: src/checkDuplicateRxn.m [hunk 4]
```diff
@@ -36,11 +37,13 @@
...
             i = i+1;
-            waitbar(i/nRxns,h);
+            %waitbar(i/nRxns,h);
+            showprogress(i/nRxns,h);
         end
         close(h);
     case 2
-		h = waitbar(0, 'Checking by reaction ...');
+		%h = waitbar(0, 'Checking by reaction ...');
+        h = showprogress(0, 'Checking by reaction ...');
```
Comment (syarra) [inline src/checkDuplicateRxn.m]: Indentation is not consistent here!

File: src/gapFilling/KEGG/mergeTwoModels.m [hunk 5]
```diff
@@ -144,10 +145,11 @@
 model2_num = length(a2);
 modelNew.S = spalloc(size(modelNew.mets,1),size(modelNew.rxns,1),model1_num+model2_num);
 
-h = waitbar(0, 'Adding Matrix 1 in Progress ...');
+%h = waitbar(0, 'Adding Matrix 1 in Progress ...');
+h = showprogress(0, 'Adding Matrix 1 in Progress ...');
 for i = 1:size(a1,1)
     modelNew.S(a1(i),b1(i)) = model1.S(a1(i),b1(i));
-    if mod(i,40) == 0,waitbar(i/size(a1,1),h);end
+    if mod(i,40) == 0,showprogress(i/size(a1,1),h);end
```
Comment (syarra) [inline src/gapFilling/KEGG/mergeTwoModels.m]: Avoid writing one-line `if`.

File: src/showprogress.m [hunk 6]
```diff
@@ -0,0 +1,25 @@
+% showprogress shows waitbars
+% Inputs
+% x, whichbar, varagin: see waitbar header for explanation
+% Feb 2017, Lemmer El Assal
```
Comment (syarra) [inline src/showprogress.m]: Docstrings should be written **after** the function declaration.

File: src/showprogress.m [hunk 7]
```diff
@@ -0,0 +1,25 @@
+% showprogress shows waitbars
```
Comment (syarra) [inline src/showprogress.m]: Move this file to the `tools` folder.

File: src/showprogress.m [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        switch WAITBAR_TYPE
+            case 1 % graphic waitbar
+                if nargin > 2
+                    fout = waitbar(x, whichbar, varargin);
+                else
+                    fout = waitbar(x, whichbar);
+                end
+            case 2 % text
+                if x > 0
+                    textprogressbar(x*100);
```
Comment (syarra) [inline src/showprogress.m]: Where is the `textprogressbar` function defined?
Add the `textprogressbar.m` file to the `external` folder (as this is not code that we wrote).

File: test/verifiedTests/testGDLS/testGDLS.m [hunk 9]
```diff
@@ -0,0 +1,36 @@
+function statusOK = testGDLS()
```
Comment (syarra) [inline test/verifiedTests/testGDLS/testGDLS.m]: This file should be removed from this PR. This is unrelated to the progress bar, Open a separate PR.

Comment (syarra) [inline test/verifiedTests/testGDLS/testGDLS.m]: Remove commits 9c6a07eab1df9997840ce70ef15a65edc2e74eca and 421625e00b8aae5b3fa8501f5eb5e0ef9d085857 from this PR.

File: src/tools/showprogress.m [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+% Output:
+%   fout: handle output from waitbar() (WAITBAR_TYPE = 1)
+%
+% .. Author: 
+%        - Lemmer El Assal (Feb 2017)
+%
+    global WAITBAR_TYPE;
+    fout = [];
+    if ~isempty(WAITBAR_TYPE)
+        switch WAITBAR_TYPE
```
Comment (laurentheirendt) [inline src/tools/showprogress.m]: @syarra , Do we even need to have the option of a standard waitbar? Would the text waitbar not be sufficient?  I believe so, as I don't see users switching back manually.

File: src/tools/showprogress.m [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            case 1 % graphic waitbar
+                if nargin > 2
+                    fout = waitbar(x, whichbar, varargin);
+                else
+                    fout = waitbar(x, whichbar);
+                end
+            case 2 % text
+                if x > 0
```
Comment (laurentheirendt) [inline src/tools/showprogress.m]: `WAITBAR_TYPE` is set to `0` in `initCobraToolbox`, shouldn't this be `case 0`?

**Comment by syarra** [comment]
@lemmerelassal, please fix this issue!
The CI reports lots of errors. One of them is:
```
14:07:49 Dynamic FBA analysis in progress ...
14:07:49 ================================================================================
14:07:49 Error occurred in testDynamicFBA/testDynamicFBA and it did not run to completion.
14:07:49 
14:07:49     ---------
14:07:49     Error ID:
14:07:49     ---------
14:07:49     'MATLAB:unassignedOutputs'
14:07:49 
14:07:49     --------------
14:07:49     Error Details:
14:07:49     --------------
14:07:49     Output argument "fout" (and maybe others) not assigned during call to "showprogress".
14:07:49     
14:07:49     Error in dynamicFBA (line 97)
14:07:49     h = showprogress(0,'Dynamic FBA analysis in progress ...');
14:07:49     
14:07:49     Error in testDynamicFBA (line 47)
14:07:49             [concentrationMatrixtest, excRxnNamestest, timeVectest, biomassVectest] = dynamicFBA(model, smi, smc, Xec, dt, time);
14:07:49 ================================================================================
```

File: src/tools/showprogress.m (hunk 10 above)
Comment (laurentheirendt) [inline src/tools/showprogress.m]: As discussed, let's leave this for now then as an option.
